kore/src/Kore/Strategies/Goal.hs

```diff
@@ -345,56 +322,60 @@ instance ClaimExtractor OnePathRule where
             _ -> Nothing
 
 instance Goal AllPathRule where
-    type Prim AllPathRule = ProofState.Prim (Rule AllPathRule)
-    type ProofState AllPathRule = ProofState.ProofState
-
     goalToRule =
         AllPathRewriteRule
         . mkRewritingRule
         . RewriteRule
         . getAllPathRule
 
-    transitionRule =
-        (withDebugProofState . transitionRuleTemplate)
-        TransitionRuleTemplate
-            { simplifyTemplate = simplify _Unwrapped
-            , removeDestinationTemplate = removeDestination _Unwrapped
-            , isTriviallyValidTemplate = isTriviallyValid _Unwrapped
-            , deriveParTemplate = deriveParAllPath
-            , deriveSeqTemplate = deriveSeqAllPath
-            }
-
-    strategy _ goals rules =
-        allPathFirstStep priorityGroups
-        :> Stream.iterate
-            id
-            ( allPathFollowupStep
-                coinductiveRewrites
-                priorityGroups
-            )
+    simplify = simplify' _Unwrapped
+    checkImplication = checkImplication' _Unwrapped
+    isTriviallyValid = isTriviallyValid' _Unwrapped
+    applyClaims claims = deriveSeqAllPath (map goalToRule claims)
+
+    applyAxioms axiomss = \goal ->
+        foldM applyAxioms1 (GoalRemainder goal) axiomss
```

Comment:
This implementation for `applyAxioms` is different from the `OnePathRule` instance because of the priority groups?

Reply:
That's right. I guess I should probably make it the same for symmetry, but when applying the rules in sequence, it doesn't make a difference anyway.

Reply:
I think it's fine the way it is, I was just making sure I understand the code properly.